src/mumble/CoreAudio.h

```diff
@@ -6,50 +6,63 @@
 #ifndef MUMBLE_MUMBLE_COREAUDIO_H_
 #	define MUMBLE_MUMBLE_COREAUDIO_H_
 
+#	include <AudioToolbox/AudioToolbox.h>
+
 #	include "AudioInput.h"
 #	include "AudioOutput.h"
 
-#	include <AudioToolbox/AudioToolbox.h>
-#	include <Carbon/Carbon.h>
-
-#	include "Global.h"
+enum AUElement { OUTPUT = 0, INPUT = 1 };
 
 class CoreAudioSystem : public QObject {
 private:
 	Q_OBJECT
 	Q_DISABLE_COPY(CoreAudioSystem)
 public:
-	static CFStringRef QStringToCFString(const QString &str);
-	static const QHash< QString, QString > getDevices(bool input);
+	static const QHash< QString, QString > getDevices(bool input, bool echo);
 	static const QList< audioDevice > getDeviceChoices(bool input);
 };
 
 class CoreAudioInput : public AudioInput {
 private:
 	Q_OBJECT
 	Q_DISABLE_COPY(CoreAudioInput)
+	// Open HAL AU as input and pass back the output stream description.
+	bool openAUHAL(AudioStreamBasicDescription &streamDescription);
+
+	// Open VoiceProcessingIO AU as input, utilizing macOS's builtin echo cancellation,
+	// and pass back the output stream description.
+	bool openAUVoip(AudioStreamBasicDescription &streamDescription);
+
+	// Initialize input AU with preferred parameters of Mumble
+	bool initializeInputAU(AudioUnit au, AudioStreamBasicDescription &streamDescription, int &actualBufferLength);
+
 protected:
-	AudioUnit au;
-	AUEventListenerRef el;
-	AudioBufferList buflist;
+	// Hardware Abstraction Layer's AudioUnit, directly interacts with the hardware
+	AudioUnit auHAL {};
+	// VoiceProcessingIO AU, provides audio input and echo cancellation
+	AudioUnit auVoip {};
+	AudioDeviceID inputDevId {};
+	AudioDeviceID echoOutputDevId {};
+	AudioBufferList buflist {};
 	static void propertyChange(void *udata, AudioUnit au, AudioUnitPropertyID prop, AudioUnitScope scope,
-							   AudioUnitElement element);
+	                           AudioUnitElement element);
```

Comment:
Did you manually change tabs to spaces here? If so: why?

Reply:
No. Just my editor went rouge.

Reply:
Okay this also happened at a few other places.
You can probably run clang-format on the source files in order to get the desired formatting.